lib/src/HttpAppFrameworkImpl.cc

```diff
@@ -348,6 +348,30 @@ PluginBase *HttpAppFrameworkImpl::getPlugin(const std::string &name)
 {
     return pluginsManagerPtr_->getPlugin(name);
 }
+void HttpAppFrameworkImpl::addPlugin(
+    const std::string& name, 
+    const std::vector<std::string>& dependencies, 
+    const Json::Value& config)
+{
+    Json::Value pluginConfig;
+    pluginConfig["name"] = name;
+    Json::Value deps(Json::arrayValue);
+    for(const auto dep : dependencies) {
+        deps.append(dep);
+    }
+    pluginConfig["dependencies"] = deps;
+    pluginConfig["config"] = config;
+    auto& plugins = jsonConfig_["plugins"];
```

Comment:
This method must be called after loading the configuration file, otherwise, the plugins added here will be overridden when the configuration file is loaded. We should declare a separate member variable to store them, and create them together with the plugins in the configuration file when the run() function is called.